src/lib/support/PersistedCounter.h

```diff
@@ -75,7 +77,30 @@ class PersistedCounter : public MonotonicallyIncreasingCounter
      *          CHIP_ERROR_INVALID_INTEGER_VALUE if aEpoch is 0.
      *          CHIP_NO_ERROR otherwise
      */
-    CHIP_ERROR Init(PersistentStorageDelegate * aStorage, KeyType aKey, uint32_t aEpoch);
+    CHIP_ERROR Init(PersistentStorageDelegate * aStorage, KeyType aKey, T aEpoch)
```

Comment:
Would like to see codesize numbers on all this inlining...

Reply:
These functions are not inlined per size report, and just checked my local size report to confirm this.

This branch (where PersistentStorageDelegate::Init is called)
```
 .text._ZN4chip6Server4InitERKNS_16ServerInitParamsE
                0x00000000002142e4      0xf7e standalone/obj/src/app/server/lib/libCHIPAppServer.a(libCHIPAppServer.Server.cpp.o)
                0x00000000002142e4                chip::Server::Init(chip::ServerInitParams const&)
```

base of this PR
```
 .text._ZN4chip6Server4InitERKNS_16ServerInitParamsE
                0x000000000016bc84      0xf7e standalone/lib/libCHIP.a(libCHIPAppServer.Server.cpp.o)
                0x000000000016bc84                chip::Server::Init(chip::ServerInitParams const&)
```

And by searching `MonotonicallyIncreasingCounter` in generated .map file, there is only one copy (as expected) in the final binary.